routes/patient/patient.js

```diff
@@ -0,0 +1,116 @@
+        .groupBy('meal.cook_id');
+
+      let resultMeals = await getMealsQuery;
+      if (!resultMeals || typeof resultMeals === 'undefined' || resultMeals.length == 0) {
+        throw new Error('Error while fetching meal details.');
+      }
+
+
+      //Getting cook details for available meals.
+
+      //Get all cook details.
+      let getCooksQuery = knex.
+        select(
+          'user.id',
+          'user.first_name',
+          'user.last_name',
+          'address.locality_id',
+          'locality.name')
+        .from('user')
+        .join('address', 'user.id', '=', 'address.user_id')
+        .join('locality', 'address.locality_id', '=', 'locality.id')
+        .where('user.user_type', '=', 'Cook');
+
+      let resultCooks = await getCooksQuery;
+
+      //Final list of available meals from cooks.
+      if (!resultCooks || typeof resultCooks === 'undefined' || resultCooks.length == 0) {
+        throw new Error('Error while fetching details.');
+      }
+
+      for (i = 0; i < resultMeals.length; i++) {
+        for (j = 0; j < resultCooks.length; j++) {
+          if (resultMeals[i].cook_id.toString() === resultCooks[j].id.toString()) {
+            resultMeals[i].cook_name = resultCooks[j].first_name + resultCooks[j].last_name;
+            resultMeals[i].locality_id = resultCooks[j].locality_id;
+            resultMeals[i].locality_name = resultCooks[j].name;
+          }
+          else
+            continue;
+        }
+      }
+
+
+      // Sort list - near and far.
+
+      //Get location details of patient.
+      if (!decodedUser) {
+        res.status(400).json({ message: 'Invalid user' });
+      } else {
+        let getUserLocalityQuery = knex.
+          select(
+            'address.locality_id')
+          .from('address')
+          .where('address.user_id', "=", decodedUser.id);
+
+        let userLocationDetails = await getUserLocalityQuery;
+
+        if (!userLocationDetails || typeof userLocationDetails === 'undefined' || userLocationDetails.length == 0) {
+          throw new Error('Error while fetching details.');
+        }
+
+        // Get service areas of patient location.
+
+        let getServiceAreas = knex.
+          select('service_areas.service_area_id')
+          .from('service_areas')
+          .where('service_areas.locality_id', '=', userLocationDetails[0].locality_id);
+
+        let serviceAreas = await getServiceAreas;
+
+        if (!serviceAreas || typeof serviceAreas === 'undefined' || serviceAreas.length == 0) {
+          throw new Error('Error while fetching details.');
+        }
+
+        // Compare cook locations with service areas of patient location. If match, nearby set to true.
+
+        for (i = 0; i < resultMeals.length; i++) {
+          for (j = 0; j < serviceAreas.length; j++) {
+            if (resultMeals[i].locality_id.toString() === serviceAreas[j].service_area_id.toString())
+            {
+              resultMeals[i].nearby = 1;
```

Comment:
Instead of using integers here for `1`s and `0`s, just use booleans. Maybe change `resultMeals[i].nearby` to `resultMeals[i].is_nearby` and set it to `true` or `false` instead of `1` and `0`. It'll make more sense for the UI also.